### PR TITLE
docs(app-of-apps): add info on app-of-apps chart structure

### PR DESCRIPTION
**Pull request by hairmare**
# Description

Add a section that explains how app-of-apps charts are structured.

# Issues
- #84

# Checklist

* [x] I updated the version in Chart.yaml
* [x] I updated applicable README.md files using  `pre-commit run`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] If I updated a dependency tool, or app, this PR contains a short summary of the changes I'm pulling
* [x] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

### CODE REVIEW

File: docs/argocd-app-of-apps.md [hunk 1]
```diff
@@ -26,7 +26,70 @@ we consider valuable on most clusters. As with all the charts, `infra-apps` can
...
+│   └── default-values.yaml
+├── examples
+│   └── example-app.yaml
+└── templates
+    ├── NOTES.txt
+    └── example-app.yaml
+```
+
+In this example the chart contains a single application called `example-app`. Most files correspond to their regular
+helm chart counterparts but there are some special files. Aapp names are dasherized in file names and camel cased
```
Comment (kebarb) [inline docs/argocd-app-of-apps.md:49]: `Aapp names` typo here

Comment (hairmare) [reply docs/argocd-app-of-apps.md]: fixed and rebased, thanks

File: docs/argocd-app-of-apps.md [hunk 2]
```diff
@@ -26,7 +26,70 @@ we consider valuable on most clusters. As with all the charts, `infra-apps` can
...
+
+`examples/example-app.yaml` contains an example on how to configure the given app using the chart:
+```yaml
+exampleApp:
+  enabled: true
+  values:
+    ingress:
+      enabled: true
+      annotations:
+        kubernetes.io/ingress.class: infra-ingress
```
Comment (kebarb) [inline docs/argocd-app-of-apps.md:70]: small correction : `kubernetes.io/ingress.class: nginx`

Comment (hairmare) [reply docs/argocd-app-of-apps.md]: Thanks for catching this one, i shall now prepare a PR to fix this everywhere else as well :stuck_out_tongue_closed_eyes:

**Review by isantospardo — APPROVED** [review]
LGTM

**Pull request merged**